rules-reviewed/eap7/eap6/resteasy.windup.xml

```diff
@@ -737,6 +737,93 @@
+                </javaclass>
+            </when>
+            <perform>
+                <iteration>
+                    <when>
+                        <not>
+                            <file-exists filename="javax.ws.rs.ext.Providers"/>
+                        </not>
+                    </when> 
+                    <perform>
+                        <hint title="Resteasy SerializableProvider is disabled by default" effort="1" category-id="mandatory">
+                            <message>Deserializing Java objects from untrusted sources is not safe. For this reason, 
+                            the org.jboss.resteasy.plugins.providers.SerializableProvider class is disabled by default, and it is not recommended to use this provider. 
+                            If you need to enable it even against the recommendation, create or update file `META-INF/services/javax.ws.rs.ext.Providers` with adding line with `org.jboss.resteasy.plugins.providers.SerializableProvider` string.
+                            </message>
+                            <link title="RESTEasy SerializableProvider"
+                                href="https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.1.beta/html-single/migration_guide/#migrate_resteasy_serializableprovider"/>
+                            <link href="http://docs.jboss.org/resteasy/docs/3.1.2.Final/javadocs/org/jboss/resteasy/plugins/providers/SerializableProvider.html" title="SerializableProvider Java API documentation"/>
+                            <tag>resteasy</tag>
+                        </hint>
+                    </perform>
+                </iteration>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000142">
+            <when>
+                <or>
+                    <javaclass references="javax.ws.rs.Produces">
+                        <location>ANNOTATION</location>
+                        <annotation-literal name="value" pattern="application/xml{*}"/>
+                    </javaclass>
+                    <javaclass references="javax.ws.rs.Produces">
+                        <location>ANNOTATION</location>
+                        <annotation-literal name="value" pattern="text/{*}"/>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <hint title="RESTEasy Text default charset response changed" effort="1" category-id="optional">
+                    <message>Resteasy should use UTF-8 as the character set for text media types.
+                    The `resteasy.add.charset` parameter is set to `true` by default. You can set the `resteasy.add.charset` parameter to false in web.xml `context-param` element.
+                    If you do not want RESTEasy to add `charset=UTF-8` to the returned content-type header when the resource method returns a `text/*` or `application/xml*` media type without an explicit charset. 
```

Comment:
Maybe it's better w/o a new sentence starting "If" but have something like:
_You can set the `resteasy.add.charset` parameter to false in web.xml `context-param` element if you do not want RESTEasy to add `charset=UTF-8` to the returned content-type header_ ...

Reply:
ok, I'll try to reword the whole message, it seems like that if you copy&paste from more resources :(